components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1.common/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/common/mappings/PublisherCommonUtils.java

```diff
@@ -891,23 +893,33 @@ public static API prepareToCreateAPIByDTO(APIDTO body, APIProvider apiProvider,
         }
 
         //Get all existing versions of  api been adding
-        List<String> apiVersions = apiProvider.getApiVersionsMatchingApiName(body.getName(), username);
+        List<String> apiVersions = apiProvider.getApiVersionsMatchingApiNameAndOrganization(body.getName(),
```

Comment:
Hope this method addresses both tenant and choreo cases.
Can we remove getApiVersionsMatchingApiName method too?

Reply:
Fixed with https://github.com/wso2/carbon-apimgt/pull/10587